discovery/pkg/sync/queue.go

```diff
@@ -27,9 +26,10 @@ import (
 )
 
 const (
-	actionAdd    = "add"
-	actionUpdate = "update"
-	actionDelete = "delete"
+	actionAdd       = "add"
+	actionUpdate    = "update"
+	actionDelete    = "delete"
+	queueMaxRetries = 5
```

Comment:
5 might be a bit high of a retry count. Maybe split this out and allow it to be configurable?

Reply:
Making it configurable is probably best. Any suggestions for default num retries? 3?

Reply:
Yeah maybe 3 is better. @rosskukulinski any thoughts?

Reply:
I've read through this PR and #134 and I'm not sure I understand this well enough to explain the impact this value might have to a user.  What are the tradeoffs of making this 4 or 6?

Is this something we expect users to have to configure frequently?  Does it depend on the size of their backend clusters (services?)

Reply:
The question is, if a service or endpoint fails to insert/update/delete, how many times should the discover retry the operation before giving up. In larger, busier clusters, I could see setting this number lower as it adds to the overhead of the queue size. 

I think 3 is a good default, then LGTM.

Reply:
Ok!  I don't see a reason to make this configurable to general users at this time, but if it's helpful for @alexbrand's performance testing/tuning, then I'm +1 for adding configuration at this time.

Otherwise, LGTM for 3 based on @stevesloka's input.